jgrapht-core/src/main/java/org/jgrapht/alg/tour/WarnsdorffRule.java

```diff
@@ -0,0 +1,873 @@
+      3). (n - 2, 2);
+      4). (1, m - 3);
+      5). (0, m - 1);
+      6). (n - 1, m - 2);
+      7). (n - 3, m - 1);
+     
+      #########################################
+      #**0***********************************2#
+      #1**************************************#
+      #*************************************3*#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #***************************************#
+      #*4*************************************#
+      #**************************************6#
+      #5***********************************7**#
+      #########################################
+      
+      Structured cells are needed to implement the merging procedure in the Parberry's algorithm.
+     */
+
+    /**
+     * Array that stores pointers on the structured cells.
+     */
+
+    private final ArrayList<DoublyLinkedList.Node<ChessBoardCell<Integer, Integer>>> structured;
+
+    /**
+     * Used in toArrayList function.
+     */
+
+    private ArrayList<ChessBoardCell<Integer, Integer>> arrayList;
+
+    public KnightTourContainer() {
+        structured = new ArrayList<>(Collections.nCopies(8, new DoublyLinkedList.Node<>()));
+        list = new DoublyLinkedList<>();
+        arrayList = null;
+    }
+
+    public ArrayList<ChessBoardCell<Integer, Integer>> toArrayList() {
+        if (arrayList != null) {
+            return arrayList;
+        }
+
+        DoublyLinkedList.Node<ChessBoardCell<Integer, Integer>> startNode = list.getStartNode();
+        startNode.setVisited(true);
+        arrayList = new ArrayList<>();
+        arrayList.add(startNode.getValue());
+
+        startNode = startNode.getNext();
+
+        arrayList.add(startNode.getValue());
+        startNode.setVisited(true);
+
+        /*
+         Traverse of the list.
+         */
+
+        while (startNode.getNext().isVisited() || startNode.getPrev().isVisited()) {
+            if (startNode.getNext().isVisited())
+                startNode = startNode.getNext();
+            else {
+                startNode = startNode.getPrev();
+            }
+            arrayList.add(startNode.getValue());
+            startNode.setVisited(true);
+        }
+
+        return arrayList;
+    }
+
+    public DoublyLinkedList<ChessBoardCell<Integer, Integer>> getList() {
+        return list;
+    }
+
+    public ArrayList<DoublyLinkedList.Node<ChessBoardCell<Integer, Integer>>> getStructured() {
+        return structured;
+    }
+}
+
+/**
+ * Implementation of the Warnsdorff's rule.
+ */
+
+public class WarnsdorffRule {
```

Comment:
Wiki calls it Warnsdorf's rule, you call it WarnsdorffRule with double ff. I'm no sure which is the correct spelling, so please double check ;)

Reply:
I've noticed that too. Wikipedia is the only source I've found that uses a single "f", all papers and other online resources use "ff", so I think it is correct.